File: 3487-maximum-unique-subarray-sum-after-deletion/3487-maximum-unique-subarray-sum-after-deletion.java
class Solution {
    public int maxSum(int[] nums) {
        Arrays.sort(nums);
        if(nums[nums.length - 1] < 0) return nums[nums.length - 1]; // if only negative numbers

        int total = 0, prev = 0;
        for(var num : nums){
            if(num < 0 || num == prev) continue;
            total += num;
            prev = num;
        }

        return total;
    }
}

File: 3487-maximum-unique-subarray-sum-after-deletion/3487-maximum-unique-subarray-sum-after-deletion.py
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        nums.sort()
        if nums[-1] < 0: return nums[-1] # if only negative numbers

        total = prev = 0
        for num in nums:
            if num < 0 or num == prev: continue
            total += num
            prev = num

        return total